lib/pry/config.rb

```diff
@@ -4,8 +4,91 @@ class Pry
   class Config < Pry::BasicObject
     include Behavior
 
+    #
+    # @return [Pry::Config]
+    #   An object who implements the default configuration for all
+    #   Pry sessions.
+    #
+    def self.defaults
+      defaults = from_hash(
+        input: Pry.lazy { lazy_readline }.memoize,
+        output: Pry.lazy { $stdout.tap { |out| out.sync = true } }.memoize,
+        commands: Pry.lazy { Pry::Commands }.memoize,
+        prompt_name: Pry.lazy { Pry::Prompt::DEFAULT_NAME }.memoize,
+        prompt: Pry.lazy { Pry::Prompt[:default] }.memoize,
+        prompt_safe_contexts: Pry.lazy { Pry::Prompt::SAFE_CONTEXTS }.memoize,
+        print: Pry.lazy { Pry::DEFAULT_PRINT }.memoize,
+        quiet: Pry.lazy { false }.memoize,
```

Comment:
1. Why can't this be just `quiet: false`?
2. We call `memoize` on every single option. This doesn't look good because it looks like default behaviour of `Pry.lazy { ... }` (I mean, why can't it call that for us?).

Reply:
1. good catch, it can and should be.

2. we don't need to memoize every default, for sure, there is actually probably very few worth memoizing. 

Because of how the config works, memoization is already there, for example:
`_pry_.config.input` and `Pry.config.input` will return a memoized object after reading from `Pry::Config.defaults`. 

Memoization would be removed if `Pry.config.forget(:input)` was called, though. 
Still I'm not actually sure we need this feature at all. wdyt?

Reply:
PTAL
https://github.com/pry/pry/pull/1897/commits/5c50a07687acd52833e5645ade1188234c29a4a6